v/V.g4

```diff
@@ -0,0 +1,1332 @@
+            this.lastToken = next;
+        }
+
+        return next;
+    }
+}
+
+//SourceFile       = ModuleClause ";" { ImportDecl ";" } { TopLevelDecl ";" } .
+sourceFile
+    : (moduleClause eos)? ( importDecl eos )* ( topLevelDecl eos )*
+    ;
+
+/*
+    V语言没有package关键字，取而代之的是 module 关键字
+*/
+
+//ModuleClause  = "module" ModuleName .
+//ModuleName    = identifier .
+moduleClause
+    : 'module' IDENTIFIER
+    ;
+
+importDecl
+    : 'import' ( importSpec  )
+    ;
+
+importSpec
+    : ( '.' | IDENTIFIER )? importPath
+    ;
+
+importPath
+    : IDENTIFIER
+    | STRING_LIT
+    ;
+
+//TopLevelDecl  = Declaration | FunctionDecl | MethodDecl .
+topLevelDecl
+    : declaration
+    | functionDecl
+    | methodDecl
+    ;
+
+//Declaration   = ConstDecl | TypeDecl | VarDecl .
+declaration
+    : constDecl
+    | varDecl
+    | structDecl
+//    | typeDecl
+    ;
+
+
+//ConstDecl      = "const" ( ConstSpec | "(" { ConstSpec ";" } ")" ) .
+constDecl
+    : 'const' '(' ( constSpec eos )* ')'
+    ;
+
+//ConstSpec      = IdentifierList [ [ Type ] "=" ExpressionList ] .
+constSpec
+    : IDENTIFIER  '=' expression
+    ;
+
+//
+//IdentifierList = identifier { "," identifier } .
+identifierList
+    : IDENTIFIER ( ',' IDENTIFIER )*
+    ;
+
+//ExpressionList = Expression { "," Expression } .
+expressionList
+    : expression ( ',' expression )*
+    ;
+
+
+
+
+
+
+////TypeDecl     = "type" ( TypeSpec | "(" { TypeSpec ";" } ")" ) .
+//typeDecl
+//    : 'type' ( typeSpec | '(' ( typeSpec eos )* ')' )
+//    ;
+//
+////TypeSpec     = identifier Type .
+//typeSpec
+//    : IDENTIFIER type
+//    ;
+//
+
+// Function declarations
+
+//FunctionDecl = "fn" FunctionName ( Function | Signature ) .
+//FunctionName = identifier .
+//Function     = Signature FunctionBody .
+//FunctionBody = Block .
+functionDecl
+    : 'fn' IDENTIFIER ( function | signature )
+    ;
+
+function
+    : signature block
+    ;
+
+//MethodDecl   = "fn" Receiver MethodName ( Function | Signature ) .
+//Receiver     = Parameters .
+methodDecl
+    : 'fn' receiver IDENTIFIER ( function | signature )
+    ;
+
+receiver
+    : parameters
+    ;
+
+/*TODO:变量定义, 语言与Go语言在定义变量的方式上有明显区别：
+  Golang: var a int = 0 (明确类型并有 var) ; var a = 0 (自动推断并有 var) ; a := 0 (自动推断没有 var) ;
+  V:      没有var关键字, 变量定义只有两种 (1) a := 0 ; (2) mut a := 0 ;
+  自2019-06-22 V语言开源到编辑该文档现在, V 并不支持多重变量定义语法
+*/
+//VarDecl    = "mut"? ( noMutDecl )
+//noMutDecl  = IdentifierList (":=" ExpressionList )
+noMutDecl
+    : identifierList ( ':=' expressionList )
+    ;
+varDecl
+    : 'mut'? ( noMutDecl )
+    ;
+
+//Block = "{" StatementList "}" .
+block
+    : '{' statementList '}'
+    ;
+
+//StatementList = { Statement ";" } .
+statementList
+    : ( statement eos )*
+    ;
+
+statement
+    : declaration
+    | simpleStmt
+    | returnStmt
+    | breakStmt
+    | continueStmt
+    | block
+    | ifStmt
+    | switchStmt
+    | forStmt
+//    | labeledStmt     //TODO: V语言没有goto故也没有对应的label实现
```

Comment:
Maybe just remove such not working code?

Reply:
Forgive my sharp words, such a terrible solution can be merged into master branch?

Now I wonder, does the other grammar rules in ANRLR been formalized proof？

Reply:
> Forgive my sharp words, such a terrible solution can be merged into master branch?

That was done not by me.

> Now I wonder, does the other grammar rules in ANRLR been formalized proof？

What do you mean by formalized proof?

Reply:
A BNF rule for a certain programming language usually represents the type system of the PL, 
we may have to prove these rules is bug free by those PL theory. At least the rule is no ambiguity.

Reply:
We only have example files that generated parsers from grammars should parse. It's far from formal proof, but it's enough for most cases.

Reply:
## My Apologize
I'm a sophomore student and just a ROOKIE in open-source community, 

@KvanTTT Thanks for all your kind suggestions 

and I would fix these AWFUL problems when I can get free from my current work.

I'm so sorry for my obtrusive PR but I have already mentioned that this PR is a RFC...

Thanks a lot for all your( @teverett ) work of failed examples founding !!


## For those grammar in comments...
@KvanTTT  Some V language grammars' implements are still on developing in V language community, and those grammars must be recorded for these language's future changes...

For example: `go` would be supported sooner, it was mentioned in the V language official documentation...

Reply:
Happy to help and thanks for your contribution to grammars-v4.